subprojects/core-api/src/main/java/org/gradle/api/configuration/BuildFeature.java

```diff
@@ -31,30 +31,43 @@
 public interface BuildFeature {
 
     /**
-     * Whether the feature was requested for the build.
+     * Status of the feature flag with three possibilities:
+     * <ul>
+     *     <li>true - requested to opt in
+     *     <li>false - requested to opt out
+     *     <li>null - no request (default behavior)
+     * </ul>
+     *
+     * Use {@link Provider#getOrNull()} to safely retrieve a nullable value or check {@link Provider#isPresent()}
+     * as the provider <b>can be undefined</b> in the case of no request.
      * <p>
-     * The provider <b>can be undefined</b> if the user did not explicitly opt in or opt out from a feature.
-     * Use {@link Provider#getOrNull()} to safely retrieve a nullable value or check {@link Provider#isPresent()}.
+     * Using Configuration Cache as an example, it's possible to opt in via {@code --configuration-cache} flag,
+     * and opt out via {@code --no-configuration-cache}.
      * <p>
-     * This method is primarily useful for gathering feature usage statistics, as it corresponds to the user intention.
+     * Note that opt-ins can be dismissed when flags for other features take precedence.
```

Comment:
🤔 I'm having troubles understanding this sentence. I _think_ it boils down to "the feature might not be enabled still, even if this provider is `true`", but this meaning of `dismiss` is beyond my English-reading skills.

I'd rephrase it into something more explicit, even if wordy, for example:
```
Note that opting into feature doesn't necessary mean the feature will be activated. 
There can be other flags that disable the feature.
For instance, running with {@code --export-keys} flag always disables Configuration Cache, 
but the BuildFeature.getRequested still returns true if {@code --configuration-cache} switch is also specified.
```

Reply:
I like this better indeed

Reply:
💅 
 
 An alternative:
 
 > Note that requesting a feature to be enabled doesn't necessarily mean the feature will be activated.
 

 ❓  Also, is the contrary also possible? IOW, are there cases where a feature becomes activated even if explicitly requested to be disabled? Say, user enables A and disables B, but since A requires B, B gets activated as well.  Or is that supposed to be considered a user error and thus causes a build failure?

Reply:
@abstratt, there are no cases currently when an explicitly disabled feature would get enabled due to something else. For CC and IP, we fail the build if CC is explicitly disabled but IP is explicitly enabled.